pkgs/desktops/lomiri/qml/lomiri-ui-toolkit/default.nix

```diff
@@ -0,0 +1,344 @@
+      "tst_listitems_itemselector.11.qml"
+      "tst_listitems_standard.11.qml"
+      "tst_listitems_standard.13.qml"
+      "tst_lomirilistview.11.qml"
+      "tst_lomiritestcase.qml"
+      "tst_multicolumnheader.13.qml"
+      "tst_optionselector.11.qml"
+      "tst_optionselector.13.qml"
+      "tst_page_with_header.13.qml"
+      "tst_pagehead_back_action.13.qml"
+      "tst_pagehead_contents_width.13.qml"
+      "tst_pagehead_sections.13.qml"
+      "tst_pagehead_visible.13.qml"
+      "tst_pageheader.13.qml"
+      "tst_pagestack.13.qml"
+      "tst_pagestack.DEPRECATED_APPHEADER_TABS.13.qml"
+      "tst_picker.11.qml"
+      "tst_picker.13.qml"
+      "tst_popover.12.qml"
+      "tst_popover.13.qml"
+      "tst_popups_dialog.13.qml"
+      "tst_popups_pagestack.13.qml"
+      "tst_pulltorefresh_pagestack_topmargin.13.qml"
+      "tst_slider.11.qml"
+      "tst_slider.13.qml"
+      "tst_switch_bug1510919.13.qml"
+      "tst_tabs.11.qml"
+      "tst_tabs.13.qml"
+      "tst_tabs.DEPRECATED_TOOLBAR.11.qml"
+      "tst_textarea.11.qml"
+      "tst_textarea_in_flickable.11.qml"
+      "tst_textfield.11.qml"
+      "tst_textinput_common.12.qml"
+      "tst_textinput_common.13.qml"
+      "tst_toggles.13.qml"
+
+      # SignalSpy on QML shader compilers are failing for unknown reasons
+      "tst_icon.11.qml"
+      "tst_icon.13.qml"
+    ]}; do
+      find tests/unit -name $badQmlTest -delete
+    done
+
+    # These tests are registered explicitly, just removing them from the list is enough
+    for badUnitTest in ${lib.strings.concatStringsSep " " [
+      # Requires a working Qt OpenGL context in ShapeMaterial
+      "inversemousearea"
+      "layouts"
+      "recreateview"
+      "subtheming"
+      "swipearea"
+
+      # Intended scaling value gets overridden since Qt 5.15.11
+      "dpr1"
+    ]}; do
+      sed -i -e "/$badUnitTest/d" tests/unit/unit.pro tests/unit/units/units.pro
+    done
+
+    # These tests don't fail explicitly, but issue warnings that the test suite doesn't expect, which it turns into errors
+    for unexpectedWarningTest in ${lib.strings.concatStringsSep "" [
+      # qml/Lomiri/Components/1.2/Toolbar.qml:96: TypeError: Cannot read property 'locked' of null
+      # I don't know QML enough to fix this, tracked upstream at https://gitlab.com/ubports/development/core/lomiri-ui-toolkit/-/issues/9
+      "tst_pagestack.DEPRECATED_TOOLBAR.11.qml"
+    ]}; do
+      sed -i -e "/EXCEPTIONS=/a '$unexpectedWarningTest' \\" tests/checkresults.sh
+    done
+  '';
+
+  # With strictDeps, QMake only picks up Qt dependencies from nativeBuildInputs
+  strictDeps = false;
+
+  nativeBuildInputs = [
+    perl
+    pkg-config
+    python3
+    qmake
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    glib
+    lttng-ust
+    qtbase
+    qtdeclarative
+    qtpim
+    qtquickcontrols2
+    qtsystems
+  ];
+
+  propagatedBuildInputs = [
+    qtfeedback
+    qtgraphicaleffects
+    qtsvg
+    qtwayland
```

Comment:
How to confirm this need propagate？

Reply:
- feedback & graphicaleffects should show up when you grep the installed qml files for import statements. they're propagated so individual applications don't need to figure out via trial-and-error which modules they depend upon also happen to require feedback / graphicaleffects
- svg... i don't remember, been too long :slightly_frowning_face:. i think for svg icon format support? Can try to check.
- wayland so all applications using the toolkit work on wayland, outside of Lomiri

Reply:
wrapQtAppsHook will propagate qtwayland in linux
https://github.com/NixOS/nixpkgs/blob/b2063364e120e14ea5a120bd144ab75a1a3b83f5/pkgs/development/libraries/qt-5/5.15/default.nix#L357

Reply:
I think it *may* have fixed usage of `lomiri.propagatedBuildInputs` for pre-check setup in some reverse dependencies actually… ehh. Building `lomiri-terminal-app` without re-propagated `qtwayland` produces wrapping that looks identical so you're right. I'm sure if whatever that was fixing comes up again, I'll notice it again. :slightly_smiling_face: